### PR TITLE
add an extremely primitive conversation caching implementation

### PR DESCRIPTION
**Pull request by bparees**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Review by vishnuchalla — APPROVED** [review]
approving this PR to help with merge conflicts problem.

File: modules/conversation_cache.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            return self.cache[key]
+        else:
+            return ""
+
+    # Adds a new key-value pair to the cache.  If the key already exists the new value will be appended to the existing value
+    def upsert(self, key, value):
+        oldVal=self.get(key)
+        if oldVal:
+            # TODO limit length of cached value
+            self.cache[key]=oldVal+"\n"+value
```
Comment (vishnuchalla) [inline modules/conversation_cache.py:30]: This operation looks a bit strange to when a key is already present. I hope it will be refined later.

Comment (bparees) [reply modules/conversation_cache.py:30]: the primary client usage currently is to append the new LLM response to the previous conversation history, hence why it behaves this way.

there's certainly no reason it couldn't be a more traditional cache in which the client is responsible for defining the new value and updating the key explicitly.

**Comment by thoraxe** [comment]
Good enough?

**Review by thoraxe — APPROVED** [review]

**Pull request merged**